### PR TITLE
GH workflows

### PR DESCRIPTION
**Pull request by digitronik**
- Moving from travis to gh-actions
- Removed py2 support and added py3.8 and py3.9 test support
- updated topics

Fixes: #19

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+jobs:
+  unit-tests:
+    # Run unit tests on different version of python
+    name: 🐍 Python-${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    needs: [ codechecks ]
+    strategy:
+      matrix:
+        python-version: [ pypy2, pypy3, 2.7, 3.6, 3.7, 3.8 ]
```
Comment (mshriver) [inline .github/workflows/unit_tests.yml:18]: I think you can drop py2 support across the board here, and we'll bump the next release to `1.2` or even `2.0`. Anyone still using py2.7 can freeze their version to `1.1.6`

@RonnyPfannschmidt @psav @quarckster any objection?

Comment (RonnyPfannschmidt) [reply .github/workflows/unit_tests.yml:18]: Yay

**Comment by mshriver** [comment]
I think you can include dropping travis since the GHA is running tox.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        run: |
+          python -m pip install --upgrade setuptools wheel twine
+          python setup.py sdist bdist_wheel
+          python -m twine check dist/*
+
+      - name: Deploy to PyPi
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.token }}
```
Comment (mshriver) [inline .github/workflows/release.yml:35]: I can add this secret when we're ready, I've got maintainer permissions on pypi.

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (digitronik) [reply .github/workflows/unit_tests.yml:18]: ahhh I nearly forgot about this PR. going to restructure it as lots of new changes in gh-workflows. Definitely better to drop py2 as per me.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - master
+    tags:
+      - '*'
+
+jobs:
+  build-and-publish:
+    name: Build and publish Python 🐍 distributions to PyPI
+    if: startsWith(github.event.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
```
Comment (quarckster) [inline .github/workflows/release.yml:15]: You can "Add a release step":

```
- name: Create a release
  uses: actions/create-release@v1
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
  with:
    tag_name: ${{ github.ref }}
    release_name: Release ${{ github.ref }}
    draft: false
    prerelease: false
```

Comment (digitronik) [reply .github/workflows/release.yml:15]: It will be great. I never tried this action before :).

File: .github/workflows/release.yml (hunk 2 above)
Comment (digitronik) [reply .github/workflows/release.yml:45]: Yup, It will great if we start adding project-level tokens.

**Comment by RonnyPfannschmidt** [comment]
i jsut took note o this, i added the sentaku workflow i #29

**Comment by digitronik** [comment]
Thanks @RonnyPfannschmidt !

**Pull request closed without merging**